utils/python/CIME/compare_test_results.py

```diff
@@ -0,0 +1,86 @@
+    expect(test_status_files, "No matching test cases found in for %s/%s/%s" % (test_root, test_id_glob, TEST_STATUS_FILENAME))
+
+    broken_compares = []
+    for test_status_file in test_status_files:
+        ts = TestStatus(test_dir=os.path.dirname(test_status_file))
+        test_name = ts.get_name()
+        if (compare_tests in [[], None] or CIME.utils.match_any(test_name, compare_tests)):
+            overall_result = ts.get_overall_test_status()
+
+            # Compute hist status, False implies it diffed
+            run_result = ts.get_status(RUN_PHASE)
+            if (run_result is None):
+                broken_compares.append((test_name, "no run phase"))
+                logging.warning("Test '%s' did not make it to run phase" % test_name)
+                hist_no_compare = True
+            elif (run_result != TEST_PASS_STATUS):
+                broken_compares.append((test_name, "test did not pass"))
+                logging.warning("Test '%s' did not pass, not safe to compare" % test_name)
+                hist_no_compare = True
+            else:
+                hist_no_compare = False
+
+            # Now, do the compare
+            if hist_no_compare:
+                logging.info("Cannot compare test: %s, overall status: %s" % (test_name, overall_result))
+            else:
+
+                logging.info("###############################################################################")
+                logging.info("Comparing results for test: %s, most recent result: %s" % (test_name, overall_result))
+                logging.info("###############################################################################")
+
+                # Get testcase dir for this test
+                if (test_id is None):
+                    # The full name already contains the compiler, so we just need to glob for the branch name
+                    globs = glob.glob("%s/%s*%s*" % (test_root, test_name, baseline_name))
+                else:
+                    globs = glob.glob("%s/%s%s" % (test_root, test_name, test_id_glob))
+
+                if (len(globs) != 1):
+                    logging.warning("Expected exactly one match for testcase area for test '%s', found '%s'" % (test_name, globs))
+                    broken_compares.append((test_name, "multiple matching testcase dirs"))
+                    continue
+
+                testcase_dir_for_test = globs[0]
+
+                # Compare hist files
+                if (not hist_no_compare):
+                    success, reason = compare_history(testcase_dir_for_test, baseline_name)
+                    if (not success):
+                        broken_compares.append((test_name, reason))
+
+    # Make sure user knows that some tests were not compareed
+    success = True
+    for broken_compare, reason in broken_compares:
+        logging.warning("COMPARE FAILED FOR TEST: %s, reason %s" % (broken_compare, reason))
```

Comment:
Ideally I would want the output from this tool to look basically the same as the output put in the TestStatus file - although just for the COMPARE_baseline phase. That is, I'd want to see something like:

PASS   SOMETEST  BASELINE_COMPARE

or

FAIL   SOMETEST  BASELINE_COMPARE

That's approximately what happened with the earlier version of these tools.

This comment applies both to the main output from the comparison, and also to various other extraneous information that is printed (or not). So, for example, I'd prefer not to have this output for every test:

```python
                logging.info("Comparing results for test: %s, most recent result: %s" % (test_name, overall_result))
```

- because I'd like to be able to parse this output in the same way that I parse the normal TestStatus or cs.status output. 

However, I agree that it does make sense to report on failed tests here. One possibility would be to print out the testStatus output for each test that failed.

Actually, maybe you could accomplish all of this cleanly by making more use of the TestStatus class here: Rather than having this function be responsible for output, it could instead just update the ts object with the results of the BASELINE_COMPARE phase (I know, there isn't a separate BASELINE_COMPARE phase yet... but the idea still applies). Then you can output the updated TestStatus results.

One complication with this is that I don't think you want to change the TestStatus file in the test itself (it seems like a Bad Idea to have some external script change the TestStatus file). In this usage, we instead would want the TestStatus to be written for each test via the logger. I think this could be accomplished by adding a __str__ method to TestStatus. This would contain the bulk of what is currently TestStatus.flush, and then TestStatus.flush could simply write out the results of str(self).